lnrpc/routerrpc/router_server.go

```diff
@@ -596,3 +600,32 @@ func (s *Server) BuildRoute(ctx context.Context,
 
 	return routeResp, nil
 }
+
+// SubscribeHtlcEvents creates a uni-directional stream from the server to
+// the client which delivers a stream of htlc events.
+func (s *Server) SubscribeHtlcEvents(req *SubscribeHtlcEventsRequest,
+	stream Router_SubscribeHtlcEventsServer) error {
+
+	htlcClient, err := s.cfg.RouterBackend.SubscribeHtlcEvents()
+	if err != nil {
+		return err
+	}
+	defer htlcClient.Cancel()
+
+	for {
+		select {
+		case event := <-htlcClient.Updates():
+			rpcEvent, err := rpcHtlcEvent(event)
+			if err != nil {
+				return err
+			}
+
+			if err := stream.Send(rpcEvent); err != nil {
+				return err
+			}
+
+		case <-s.quit:
```

Comment:
I see we in other places also have `case <-stream.Context().Done():`, in case the client disconnects

Reply:
Also noticed that `trackPayment` doesn't listen to a quit signal, maybe worth adding that one-liner.